File: src/main/java/com/tomaszezula/stripe101/config/StripeConfig.java
package com.tomaszezula.stripe101.config;

import com.stripe.Stripe;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;

@Configuration
public class StripeConfig {

    public StripeConfig(@Value("${stripe.secret-key}") String secretKey) {
        Stripe.apiKey = secretKey;
    }
}

File: src/main/kotlin/com/tomaszezula/stripe101/config/StripeConfig.kt
package com.tomaszezula.stripe101.config

import com.stripe.Stripe
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Configuration

@Configuration
@EnableConfigurationProperties(CheckoutSessionConfig::class)
class StripeConfig(@Value("\${stripe.secret-key}") private val secretKey: String) {
    init {
        Stripe.apiKey = secretKey
    }

}